lib/harvester.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = class Harvester {
```

Comment:
I'd suggest a jsdoc block that describes what this container is, what it replaces, and just the general overview of how it is intended to work.